src/EditorFeatures/Core/Extensibility/NavigationBar/AbstractEditorNavigationBarItemService.cs

```diff
@@ -77,30 +84,16 @@ public Task<bool> TryNavigateToItemAsync(Document document, NavigationBarItem it
                 // Otherwise, the item pointed to a location in another document.  Just return the position we
                 // computed and stored for it.
                 Contract.ThrowIfNull(symbolItem.Location.OtherDocumentInfo);
-                var otherLocation = symbolItem.Location.OtherDocumentInfo.Value;
-                return Task.FromResult((otherLocation.documentId, otherLocation.navigationSpan.Start, 0));
+                var (documentId, navigationSpan) = symbolItem.Location.OtherDocumentInfo.Value;
+                return Task.FromResult((documentId, navigationSpan.Start, 0));
             }
         }
 
-        protected void NavigateToVirtualTreePoint(Solution solution, VirtualTreePoint navigationPoint, CancellationToken cancellationToken)
+        public bool ShowItemGrayedIfNear(NavigationBarItem item)
         {
-            this.AssertIsForeground();
-            var documentToNavigate = solution.GetRequiredDocument(navigationPoint.Tree);
-            var workspace = solution.Workspace;
-            var navigationService = workspace.Services.GetRequiredService<IDocumentNavigationService>();
-
-            if (navigationService.CanNavigateToPosition(workspace, documentToNavigate.Id, navigationPoint.Position, navigationPoint.VirtualSpaces, cancellationToken))
-            {
-                navigationService.TryNavigateToPosition(workspace, documentToNavigate.Id, navigationPoint.Position, navigationPoint.VirtualSpaces, options: null, cancellationToken);
-            }
-            else
-            {
-                var notificationService = workspace.Services.GetRequiredService<INotificationService>();
-                notificationService.SendNotification(EditorFeaturesResources.The_definition_of_the_object_is_hidden, severity: NotificationSeverity.Error);
-            }
```

Comment:
had this code already.  So removed the duplicated method and had the caller just call into the otehr impl.